specification/k3po.control/SPEC.md

```diff
@@ -240,3 +217,50 @@ This event is sent by the driver when the driver encountered an error while proc
 The message body for the event is the description of the exception in case we have an exception or a more detailed description in case
 there is no exception. 
 
+
+## Message Exchanges
+
+The test framework initializes a connection to the driver.  Only one test may be in execution at a time per a single 
+connection.  
+
+The test framework initiates the start of a test with a **PREPARE** command. When receiving a **PREPARE** the driver SHOULD 
+load and prepare a set of scripts for execution.   If there are `accept` instructions in the scripts, a bind on 
+corresponding ports/files MUST be performed during preparation.
+
+When the scripts are successfully prepared, the driver MUST send a **PREPARED** event.
+
+The test framework signals the start of a test with the **START** command.  The driver MUST start the script execution
+when it receives the **START** command.
+
+The driver MUST send the **STARTED** event when the scripts have started.
+
+The driver MUST send the **FINISHED** event, when
+  1) all K3PO scripts have finished, either by successfully completing or 
+hitting undefined behavior AND 
+  2) after sending a **STARTED** event (note, script execution may have completed if the script
+is just accepts prior to the **START** command or **STARTED** event)
+
+Any time after the **PREPARE** command and before the **DISPOSE** command, the test framework can send an **AWAIT** command.
+ 
+If the driver receives an **AWAIT** command and the scripts are still executing (i.e. the **FINISHED** command has not been sent
+ or is not in the process of being sent), the driver MUST send a **NOTIFIED** event if the barrier has already complete or 
+ when/if the barrier does completes.  The driver MAY ignore await commands if it receives them after script execution.
+
+Prior to receiving a **FINISHED** command, the test framework may request script execution to stop by sending an  **ABORT** 
+command. The driver must abort the current script execution, which should trigger a **FINISHED** command.  If the script 
+execution has already stopped the driver will ignore the **ABORT** command (note, in this case the driver should already have
+sent the **FINISHED** or be in the process of sending the finished). 
+
+The test framework signals the end of the test by sending a **DISPOSE** command to the client.  Note the test can not end 
+prior to completing the script execution (i.e. having sent a **FINISHED** event).  This triggers the clean up of any 
```

Comment:
> Also, dispose is executing an abort on the Robot, it is not waiting for the scripts to finish.

I don't believe that is the purpose of dispose nor what I have written here.  I fixed the client/driver typo.  The scripts, successfully completed or not, can leave client or server connections / bindings open.  The dispose is to notify when it safe for those connections to be closed.   I've reworded to reinforce and clarify this meaning.  please review